### PR TITLE
Holesky support

### PR DESCRIPTION
**Pull request by prestonvanloon**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**

Adds holesky config support for the upcoming holesky testnet.

The holesky genesis state is far too large to bundle inside of the prysm distribution binary (200Mb) so it must be loaded from disk or checkpoint sync. This is existing behavior for other testnets, but I felt it would be helpful to provide a meaningful failure message when the user attempts to run prysm on a non-mainnet network. 

**Which issues(s) does this PR fix?**

#12724

**Other notes for review**

Example of running Prysm without a genesis state

```
[2023-08-28 11:37:34]  WARN flags: Running on the Holesky Beacon Chain Testnet
[2023-08-28 11:37:34]  WARN node: In order to receive transaction fees from proposing blocks, you must provide flag --suggested-fee-recipient with a valid ethereum address when starting your beacon node. Please see our documentation for more information on this requirement (https://docs.prylabs.network/docs/execution-node/fee-recipient).
[2023-08-28 11:37:34]  INFO node: Checking DB database-path=/tmp/data/beaconchaindata
[2023-08-28 11:37:34]  INFO db: Opening Bolt DB at /tmp/data/beaconchaindata/beaconchain.db
[2023-08-28 11:37:34]  INFO node: Deposit contract: 0x4242424242424242424242424242424242424242
[2023-08-28 11:37:34] ERROR node: No genesis block/state is found. Prysm only provides a mainnet genesis state bundled in the application. You must provide the --genesis-state or --genesis-beacon-api-url flag to load a genesis block/state for this network.
unable to start beacon node: could not get genesis block root
OriginGenesisRoot: not found in db
```

### CODE REVIEW

File: beacon-chain/node/node.go [hunk 1]
```diff
@@ -212,6 +212,11 @@ func New(cliCtx *cli.Context, opts ...Option) (*BeaconNode, error) {
 
 	log.Debugln("Starting State Gen")
 	if err := beacon.startStateGen(ctx, bfs, beacon.forkChoicer); err != nil {
+		if errors.Is(err, stategen.ErrNoGenesisBlock) {
+			log.Errorf("No genesis block/state is found. Prysm only provides a mainnet genesis "+
+				"state bundled in the application. You must provide the --%s or --%s flag to load "+
+				"a genesis block/state for this network.", "genesis-state", "genesis-beacon-api-url")
```
Comment (kasey) [inline beacon-chain/node/node.go:218]: I see you used variable interopolation for the flag names - did you intend to replace these with direct flag var .Name references?

File: cmd/beacon-chain/main.go [hunk 2]
```diff
@@ -288,7 +288,7 @@ func startNode(ctx *cli.Context) error {
 
 	beacon, err := node.New(ctx, opts...)
 	if err != nil {
-		return err
+		return fmt.Errorf("unable to start beacon node: %w", err)
```
Comment (kasey) [inline cmd/beacon-chain/main.go:291]: This will break wrapping, would it be better to use errors.Wrap here?

**Review by kasey — APPROVED** [review]
Some nitpicks but otherwise LGTM

File: config/features/config.go [hunk 3]
```diff
@@ -154,6 +161,10 @@ func applyPraterFeatureFlags(ctx *cli.Context) {
 func applySepoliaFeatureFlags(ctx *cli.Context) {
 }
 
+// Insert feature flags within the function to be enabled for Holesky testnet.
+func applyHoleskyFeatureFlags(ctx *cli.Context) {
```
Comment (james-prysm) [inline config/features/config.go:165]: i think deepsource is complaining these parameters are not used

File: config/params/BUILD.bazel [hunk 4]
```diff
@@ -55,6 +57,7 @@ go_test(
         "@consensus_spec_tests_mainnet//:test_data",
         "@consensus_spec_tests_minimal//:test_data",
         "@eth2_networks//:configs",
+        "@holesky_testnet//:configs",
```
Comment (james-prysm) [inline config/params/BUILD.bazel:60]: is prater handled differently as there doesn't seem to be a prater_testnet//:configs

File: beacon-chain/node/node.go (hunk 1 above)
Comment (prestonvanloon) [reply beacon-chain/node/node.go:218]: I tried this and got hit with a circular dep. I would have to refactor / move those flags to another package so I just hard coded it in the laziest way. I could remove the templating if that's your feedback here.

File: config/params/BUILD.bazel (hunk 4 above)
Comment (prestonvanloon) [reply config/params/BUILD.bazel:60]: The author that added prater did not add any tests for it. I've added this to check our config is correct in unit test

Comment (prestonvanloon) [reply config/params/BUILD.bazel:60]: oh nevermind, they did add tests. It's part of the `@eth2_networks`

File: cmd/beacon-chain/main.go (hunk 2 above)
Comment (prestonvanloon) [reply cmd/beacon-chain/main.go:291]: @kasey I don't think it breaks wrapping. You can unwrap an error that was templated with `%w`. https://pkg.go.dev/errors#example-Unwrap

Comment (kasey) [reply cmd/beacon-chain/main.go:291]: Oh my mistake - TIL the ` %w` verb, very cool!

File: beacon-chain/node/node.go (hunk 1 above)
Comment (kasey) [reply beacon-chain/node/node.go:218]: Makes sense, in that case, yes the templating seems to introduce a smidge of confusion for no discernible benefit, but I don't feel strongly about it.

File: config/features/config.go (hunk 3 above)
Comment (prestonvanloon) [reply config/features/config.go:165]: I think it's fine. These are essentially reserved functions for future use anyway

**Comment by prestonvanloon** [comment]
> This looks good to me
> 
> @prestonvanloon will you be updating the prysm docs? if not, we should get an owner on that task?

@terencechain -- yes, i will update docs as well. https://github.com/prysmaticlabs/prysm/issues/12724 as assigned to me and we have a task item there.

**Pull request merged**